components/core/src/clp/clo/OutputHandler.hpp

```diff
@@ -51,10 +54,39 @@ class OutputHandler {
     }
 };
 
+class NetworkOutputHandler : public OutputHandler {
+public:
+    // Types
+    class OperationFailed : public TraceableException {
+    public:
+        // Constructors
+        OperationFailed(ErrorCode error_code, char const* const filename, int line_number)
+                : TraceableException(error_code, filename, line_number) {}
+
+        // Methods
+        char const* what() const noexcept override { return "NetworkClient operation failed"; }
+    };
+
+    // Constructors
+    NetworkOutputHandler(std::string const& host, std::string const& port);
```

Comment:
Because eventually we need to convert int to string, and clp-s converts it when constructing `NetworkOutputHandler`

Reply:
I see, didn't realize clp-s did the same. Can we move it into the constructor for both `clo` and `clp-s`? I prefer to leave the conversion until absolutely necessary since that allows easier validation, it's less passing around of objects, and it's clearer imo.